test/index.js

```diff
@@ -1,41 +1,47 @@
-var should = require('should');
+var should = require('should'),
+    superagent = require('superagent');
 
 describe('superagent-use', function() {
-  var superagent;
+  var request;
+
   var prefix = 'http://example.com';
-  var prefixMiddleware = function(request) {
-    if(request.url[0] === '/') {
-      request.url = prefix + request.url;
+  var prefixMiddleware = function(req) {
+    if(req.url[0] === '/') {
+      req.url = prefix + req.url;
     }
-    return request;
+    return req;
   };
 
   beforeEach(function() {
-    /* superagent-use modifies the superagent object, so we need to remove
-       the `require` cache in order to get a new instance for each test. */
-    delete require.cache[require.resolve('superagent')];
-    superagent = require('superagent');
-    require('..')(superagent);
+    request = require('..')(superagent);
```

Comment:
Now that we're no longer going to use `superagent` as variable for the new instance that should be fine. But multiple calls to `require('superagent')` shouldn't impact performance or anything, because, like you showed before, modules are cached.